browser/ai_chat/ai_chat_api_request.cc

```diff
@@ -0,0 +1,115 @@
+      semantics {
+        sender: "AI Chat"
+        description:
+          "This is used to communicate with our partner API"
+          "on behalf of the user interacting with the ChatUI."
+        trigger:
+          "Triggered by user sending a prompt."
+        data:
+          "Will generate a text completion that attempts to match whatever context or pattern the user gave it"
+        destination: WEBSITE
+      }
+      policy {
+        cookies_allowed: NO
+        policy_exception_justification:
+          "Not implemented."
+      }
+    )");
+}
+
+GURL GetURLWithPath(const std::string& host, const std::string& path) {
+  return GURL(std::string(url::kHttpsScheme) + "://" + host).Resolve(path);
+}
+
+std::string CreateJSONRequestBody(base::ValueView node) {
+  std::string json;
+  base::JSONWriter::Write(node, &json);
+  return json;
+}
+
+}  // namespace
+
+AIChatAPIRequest::AIChatAPIRequest(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : api_request_helper_(GetNetworkTrafficAnnotationTag(),
+                          url_loader_factory) {}
+
+AIChatAPIRequest::~AIChatAPIRequest() = default;
+
+void AIChatAPIRequest::QueryPrompt(ResponseCallback callback,
+                                   const std::string& prompt) {
+  auto internal_callback =
+      base::BindOnce(&AIChatAPIRequest::OnGetResponse,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  base::Value::Dict dict;
+  base::Value::List stop_sequences;
+  stop_sequences.Append("\n\nHuman:");
+
+  dict.Set("prompt", prompt);
+  dict.Set("max_tokens_to_sample", 200);
+  dict.Set("temperature", 0.7);
+  dict.Set("top_k", -1);
+  dict.Set("top_p", 0.999);
+  dict.Set("model", "claude-v1");
+  dict.Set("stop_sequences", std::move(stop_sequences));
+  dict.Set("stream", false);
+
+  base::flat_map<std::string, std::string> headers;
+  headers.emplace("x-brave-key", BUILDFLAG(BRAVE_SERVICES_KEY));
+
+  VLOG(1) << __func__ << " Prompt: " << prompt << "\n";
+
+  api_request_helper_.Request("POST",
+                              GetURLWithPath(BUILDFLAG(BRAVE_AI_CHAT_ENDPOINT),
+                                             ai_chat::kAIChatCompletionPath),
+                              CreateJSONRequestBody(dict), "application/json",
+                              true, std::move(internal_callback), headers);
+
+  VLOG(1) << __func__ << " API Request sent\n";
+}
+
+void AIChatAPIRequest::OnGetResponse(
+    ResponseCallback callback,
+    api_request_helper::APIRequestResult result) {
+  const bool success = result.response_code() == 200;
+
+  if (!success) {
+    VLOG(1) << __func__ << " Response from API was not HTTP 200 (Received "
+            << result.response_code() << ")";
+    return;
+  }
+
+  std::string response = result.body();
+  const std::string* completion =
+      result.value_body().GetDict().FindString("completion");
+
+  if (completion) {
+    response = *completion;
```

Comment:
This may or may not be the best place to do this, but we should strip any HTML tags from the server response since we don't expect/want them.

Reply:
It's worth considering the user expectations here. For example, if a user asks a question about how to format text in HTML, removing HTML tags from the response might not provide the expected information. Therefore, we should be mindful of creating controlled environments balancing with user expectations.

Reply:
If we are expecting HTML tags in answers, then our options are to sanitize the HTML (to remove any scripting) or to escape the tags (if we just want the tags themselves to be visible in a StackOverflow-style use-case).

Reply:
@fmarier so is it acceptable to **not** strip any html and always treat the incoming string as text content. i.e. for desktop's WebUI, always escape the html via React's child prop (a.k.a `innerText`)? Trusted Types should prevent the content being set as innerHTML anywhere, as far as I understand.

Reply:
If it's already escaped later on, then I guess it's fine.